src/read.c

```diff
@@ -2006,6 +2023,107 @@ static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata, cons
 }
 #endif // AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
+static avifResult avifParseSampleTransformTokens(avifROStream * s, avifSampleTransformExpression * expression)
+{
+    uint8_t tokenCount;
+    AVIF_CHECK(avifROStreamRead(s, &tokenCount, /*size=*/1)); // unsigned int(8) token_count;
+    AVIF_CHECKERR(avifArrayCreate(expression, sizeof(expression->tokens[0]), tokenCount), AVIF_RESULT_OUT_OF_MEMORY);
+
+    for (uint32_t t = 0; t < tokenCount; ++t) {
+        avifSampleTransformToken * token = (avifSampleTransformToken *)avifArrayPush(expression);
+        AVIF_ASSERT_OR_RETURN(token != NULL);
+
+        AVIF_CHECK(avifROStreamRead(s, &token->value, /*size=*/1)); // unsigned int(8) token;
+        if (token->value == AVIF_SAMPLE_TRANSFORM_CONSTANT) {
+            // TODO(yguyon): Verify two's complement representation is guaranteed here.
+            uint32_t constant;
+            AVIF_CHECK(avifROStreamReadU32(s, &constant)); // signed int(1<<(bit_depth+3)) constant;
+            token->constant = *(int32_t *)&constant;       // maybe =(int32_t)constant; is enough
```

Comment:
Yes, `=(int32_t)constant;` is enough. Alternatively, we can do this:

```
    AVIF_CHECK(avifROStreamReadU32(s, (uint32_t *)&token->constant)); // signed int(1<<(bit_depth+3)) constant;
```

Reply:
> Yes, `=(int32_t)constant;` is enough.

https://en.cppreference.com/w/c/language/conversion#Integer_conversions seems to say it is implementation-defined. Should we consider it as widespread as two's complement representation?

> Alternatively, we can do this:
> 
> ```
>     AVIF_CHECK(avifROStreamReadU32(s, (uint32_t *)&token->constant)); // signed int(1<<(bit_depth+3)) constant;
> ```

I do not think it is equivalent depending on the platform's endianness. Should we consider it as widespread as two's complement representation?

Reply:
The reason casting a `uint32_t` value greater than `INT32_MAX` to `int32_t` is implementation-defined is that there are three representations of signed integers in C. In practice, the only widely-used repesentation is two's complement.